neuraxle/metaopt/auto_ml.py

```diff
@@ -878,38 +884,49 @@ def kfold_cross_validation_split(k_fold: int):
     :param k_fold: number of folds.
     :return:
     """
-    def split(data_inputs: Iterable, expected_outputs: Iterable):
-        data_inputs_train, data_inputs_val = _kfold_cross_validation_split(data_inputs)
+
+    def __init__(self, k_fold: int):
+        BaseValidationSplitter.__init__(self)
+        self.k_fold = k_fold
+
+    def split(self, data_inputs, expected_outputs=None) -> Tuple[List, List, List, List]:
```

Comment:
Maybe `def call` instead? This would allow for passing functions as well as splitters. Idk.

Reply:
I have tried to do this, and failed. I think it's fine like this tbh